Documentation/ceph-fs-subvolumegroup.md

```diff
@@ -0,0 +1,44 @@
+---
+title: CephFilesystemSubVolumeGroup CRD
+weight: 3610
+indent: true
+---
+
+{% include_relative branch.liquid %}
+
+This guide assumes you have created a Rook cluster as explained in the main [Quickstart guide](quickstart.md)
+
+# CephFilesystemSubVolumeGroup CRD
```

Comment:
Another thought... Do we really need to define a subvolume group in CRDs? Instead, this feels like it could be defined in the storage class. The admin could define the subvolume group name and also quotas in the storage class. This would just require the CSI driver to handle the subvolume group creation the first time it is requested. For RBD namespaces it feels similar where the storage class could define it. However, for external clusters if Rook doesn't have a keyring to create subvolume groups or rbd namespaces, the csi driver approach wouldn't work.

Reply:
The problem is that the SC is immutable and the quotas are subject to change over time. Also, the CSI driver does not always have the SC details (that's what @Madhu-1 told me). For external mode, the expectation is really to have the subvolumegroup present already. For another external scenario (k8s to k8s), this is handled differently and that's why we would have the CRD.

Reply:
We cannot add the subvolumegroup to the storageclass I explained the problem of adding it to storageclass here https://github.com/ceph/ceph-csi/issues/1884#issuecomment-787665156. This should be added to the csi configmap. Once we have the CRD change we need to work on updating the configmap to have new clusterID as i mentioned here  https://github.com/rook/rook/issues/7036#issuecomment-987568635